File: Projects/0_Message_Receiver/python/udp_receiver.py
import socket

def receiver():
    localhost = '127.0.0.1'
    port = 6913
    sck = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    sck.bind((localhost, port))

    data = None
    print('Listening for messages')
    while data != b'exit':
        data, _address = sck.recvfrom(1024)
        print(data.decode())

def sender():
    localhost = '127.0.0.1'
    port = 6913
    sck = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    
    message = None
    while message != 'exit':
        message = input('Insert message to send (send `exit` to stop): ')
        sck.sendto(message.encode('utf-8'), (localhost, port))



if __name__ == '__main__':
    option = input('[send] or [receive] data? ')
    match option:
        case 'send':
            sender()
        case 'receive':
            receiver()
        case _:
            exit()

File: Projects/0_Message_Receiver/go/udp_receiver.go
package main

import (
	"fmt"
	"net"
	"os"
	"strings"
)

func receiver() error {
	listener, err := net.ListenPacket("udp4", ":6913")
	if err != nil {
		return err
	}
	defer listener.Close()

	dataBuff := make([]byte, 1024)
	for {
		n, _, err := listener.ReadFrom(dataBuff)
		if err != nil {
			return err
		}
		message := strings.TrimSpace(string(dataBuff[:n]))
		if message == "exit" {
			break
		}

		fmt.Println(message)
	}

	return nil
}

func sender() error {
	socket, err := net.Dial("udp4", "127.0.0.1:6913")
	if err != nil {
		return err
	}
	defer socket.Close()

	var message []byte

	for {
		fmt.Scan(&message)
		_, err = socket.Write(message)
		if err != nil {
			return err
		}

		if string(message) == "exit" {
			break
		}
	}

	return nil
}

func main() {
	var option string
	fmt.Println("[send] or [receive] data?")
	fmt.Scan(&option)
	switch option {
	case "send":
		err := sender()
		if err != nil {
			fmt.Println(err)
		}
	case "receive":
		err := receiver()
		if err != nil {
			fmt.Println(err)
		}
	default:
		os.Exit(0)
	}
}
